config/300-pipeline.yaml

```diff
@@ -0,0 +1,91 @@
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: pipelines.messaging.knative.dev
+  labels:
+    knative.dev/crd-install: "true"
+spec:
+  group: messaging.knative.dev
+  version: v1alpha1
+  names:
+    kind: Pipeline
+    plural: pipelines
+    singular: pipeline
+    categories:
+    - all
+    - knative
+    - eventing
+  scope: Namespaced
+  subresources:
+    status: {}
+  additionalPrinterColumns:
+    - name: Ready
+      type: string
+      JSONPath: ".status.conditions[?(@.type==\"Ready\")].status"
+    - name: Reason
+      type: string
+      JSONPath: ".status.conditions[?(@.type==\"Ready\")].reason"
+    - name: Age
+      type: date
+      JSONPath: .metadata.creationTimestamp
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          required:
+            - steps
+            - channelTemplate
+          properties:
+            steps:
+              type: array
+              items:
+              properties:
+                dnsName:
+                  type: string
+                  minLength: 1
+                uri:
+                  type: string
+                  minLength: 1
+                ref:
+                  type: object
+                  required:
+                    - apiVersion
+                    - kind
+                    - name
+                  properties:
+                    apiVersion:
+                      type: string
+                      minLength: 1
+                    kind:
+                      type: string
+                      minLength: 1
+                    name:
+                      type: string
+                      minLength: 1
+            channelTemplate:
+              type: object
+              required:
+                - apiVersion
+                - kind
+              properties:
+                apiVersion:
+                  type: string
+                  minLength: 1
+                kind:
+                  type: string
+                  minLength: 1
+                spec:
+                  type: object
```

Comment:
How does this get validated?

Reply:
you can't in the openapi doc. You could if you loaded the channel template crd. That can be done in the future.

Reply:
Yeah, so it should fail now at CR creation time, so while we can't easily ( as Scott said) validate the target channel, at least when we try to create it will fail instead of previous provisioner model where the channel would get created and then fail at a later time. I'd like to not tackle this as part of this PR. One option would be to use dry run. Regardless, created this issue to track that, ok?
https://github.com/knative/eventing/issues/1349